parlai/core/params.py

```diff
@@ -890,7 +890,10 @@ def parse_args(self, args=None, namespace=None, print_args=True):
         return ``None``.
         """
         self.add_extra_args(args)
-        self.args = super().parse_args(args=args)
+        if force_known:
```

Comment:
i would feel more comfortable if there was instead a special `parse_known_args_with_postprocessing` that had the behavior you're seeking (force_known=False), and parse_args() keeps it's current behavior. Refactoring out the custom post-parsing would then be the necessary bit.